src/Microsoft.Health.Fhir.Shared.Api/Modules/HealthChecks/HealthCheckConfiguration.cs

```diff
@@ -0,0 +1,49 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using EnsureThat;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Health.Fhir.Api.Modules.HealthChecks
+{
+    public class HealthCheckConfiguration : IPostConfigureOptions<HealthCheckServiceOptions>
+    {
+        private readonly IServiceProvider _serviceProvider;
+
+        public HealthCheckConfiguration(IServiceProvider serviceProvider)
+        {
+            EnsureArg.IsNotNull(serviceProvider, nameof(serviceProvider));
+
+            _serviceProvider = serviceProvider;
+        }
+
+        public void PostConfigure(string name, HealthCheckServiceOptions options)
+        {
+            var list = options.Registrations.ToArray();
```

Comment:
nit: coding style says only use var if right hand-size is clear (here and below in foreach).